controllers/rabbitmq/transporturl_controller.go

```diff
@@ -0,0 +1,270 @@
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			rabbitmqv1beta1.TransportURLReadyCondition,
+			condition.RequestedReason,
+			condition.SeverityInfo,
+			rabbitmqv1beta1.TransportURLInProgressMessage))
+		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+
+	// TODO(dprince): Future we may want to use vhosts for each OpenStackService instead.
+	// vhosts would likely require use of https://github.com/rabbitmq/messaging-topology-operator/ which we do not yet include
+	username, ctrlResult, err := secret.GetDataFromSecret(ctx, helper, rabbit.Status.DefaultUser.SecretReference.Name, 10, "username")
+	if err != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			rabbitmqv1beta1.TransportURLReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			rabbitmqv1beta1.TransportURLReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	} else if (ctrlResult != ctrl.Result{}) {
+		return ctrlResult, nil
+	}
+
+	password, ctrlResult, err := secret.GetDataFromSecret(ctx, helper, rabbit.Status.DefaultUser.SecretReference.Name, 10, "password")
+	if err != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			rabbitmqv1beta1.TransportURLReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			rabbitmqv1beta1.TransportURLReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	} else if (ctrlResult != ctrl.Result{}) {
+		return ctrlResult, nil
+	}
+
+	host, ctrlResult, err := secret.GetDataFromSecret(ctx, helper, rabbit.Status.DefaultUser.SecretReference.Name, 10, "host")
+	if err != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			rabbitmqv1beta1.TransportURLReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			rabbitmqv1beta1.TransportURLReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	} else if (ctrlResult != ctrl.Result{}) {
+		return ctrlResult, nil
+	}
+
+	// Create a new secret with the transport URL for this CR
+	secret := r.createTransportURLSecret(instance, string(username), string(password), string(host))
+	_, op, err := oko_secret.CreateOrPatchSecret(ctx, helper, instance, secret)
+	if err != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			rabbitmqv1beta1.TransportURLReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			rabbitmqv1beta1.TransportURLReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	}
+	if op != controllerutil.OperationResultNone {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			rabbitmqv1beta1.TransportURLReadyCondition,
+			condition.RequestedReason,
+			condition.SeverityInfo,
+			rabbitmqv1beta1.TransportURLReadyInitMessage))
+		return ctrl.Result{RequeueAfter: time.Second * 5}, nil
+	}
+
+	// Update the CR and return
+	instance.Status.SecretName = secret.Name
+	if err := r.Client.Status().Update(ctx, instance); err != nil {
+		return ctrl.Result{}, err
+	}
```

Comment:
I wonder if this is necessary with the `defer()` status func in the main reconcile loop?  It `Patch`es the status there, so it should catch and apply the `SecretName` change

Reply:
good call, let me try it